src/app/workflow/view/view.service.ts

```diff
@@ -134,8 +177,11 @@ export class ViewService {
     if (selectedWorkflow.id === workflow.id) {
       this.workflowService.setWorkflow(workflow);
     }
+
+    const workflowName: string = workflow.workflowName || workflow.repository;
```

Comment:
This could have been a method, so if/when you implement Andrew's suggestion, you only have to do it once.